docs/topics/tasks.txt

```diff
@@ -0,0 +1,337 @@
+
+These arguments correspond to attributes on the created
+:class:`Task <django.tasks.Task>`.
+
+By convention, tasks should be defined in a ``tasks.py`` file, however this is
+not enforced.
+
+Modifying tasks
+---------------
+
+Before enqueueing tasks, it may be necessary to modify certain parameters of the task.
+For example, to give it a higher priority than it would normally.
+
+A ``Task`` instance cannot be modified directly. Instead, a modified instance
+can be created with the :meth:`using <django.tasks.Task.using>` method, leaving
+the original as-is::
+
+    calculate_meaning_of_life.priority  # 0
+
+    calculate_meaning_of_life.using(priority=10).priority  # 10
+
+``using`` allows modifying the following attributes:
+
+* :attr:`priority <django.tasks.Task.priority>`
+* :attr:`backend <django.tasks.Task.backend>`
+* :attr:`queue_name <django.tasks.Task.queue_name>`
+* :attr:`run_after <django.tasks.Task.run_after>`
+
+``run_after`` may also be provided as a :class:`timedelta <datetime.timedelta>`,
+which is used relative to the current time (when ``using`` is called), or a
+timezone-aware :class:`datetime <datetime.datetime>`.
+
+Enqueueing tasks
+================
+
+To add the task to the queue store, so it will be executed, call the ``enqueue``
+method on it::
+
+    result = calculate_meaning_of_life.enqueue()
+
+Returned is a :class:`django.tasks.TaskResult`, which can be used to retrieve
+the result of the task once it has finished executing.
+
+If the task takes arguments, these can be passed as-is to ``enqueue``::
+
+    result = calculate_meaning_of_life.enqueue(answer=42)
+
+To enqueue tasks in an ``async`` context, :meth:`aenqueue <django.tasks.Task.aenqueue>`
+is available as an ``async`` variant of ``enqueue``.
+
+Transactions
+------------
+
+By default, tasks are enqueued after the current transaction (if there is one)
+commits successfully (using :meth:`transaction.on_commit <django.db.transaction.on_commit>`),
+rather than enqueueing immediately.
+
+This behavior can be changed by changing the :setting:`TASKS-ENQUEUE_ON_COMMIT`
+setting for the backend, or for a specific task using the ``enqueue_on_commit``
+parameter.
+
+Task results
+============
+
+When enqueueing a ``Task``, you receive a :class:`django.tasks.TaskResult`,
+however it's likely useful to retrieve the result from somewhere else (for example
+another request or another task).
+
+Each ``TaskResult`` has a unique :attr:`id <django.tasks.TaskResult.id>`, which
+can be used to identify and retrieve the result once the code which enqueued the
+task has finished.
+
+The :meth:`django.tasks.Task.get_result` method can retrieve a result based on
+its ``id``::
+
+    # Later, somewhere else...
+    result = calculate_meaning_of_life.get_result(result_id)
+
+If the ``TaskResult`` exists, it is returned. If it doesn't exist, or isn't a
+result for ``calculate_meaning_of_life``, :exc:`django.tasks.exceptions.ResultDoesNotExist`
```

Comment:
That could work, however I intentionally left it out in case of future optimisation. I think of it like using the same integer id for two different DB tables.

Reply:
I think if you want to have a version without future optimizations, then you should say "a ``TaskResult`` with the matching ID is returned if it exists.". That phrasing removes any questions of tasks from different results.

I think we shouldn't mention the case of looking up a result from a different task, because no matter what `get_result` _should never return a result from a different task_. It can look up a result for the task it is a method of. 

Of course backends will need to check that the result is for the right task when we just have an ID and are sharing result storage across tasks, but it's an implementation detail of the backend. I would prefer having a stricter error if globally unique IDs are being used for a backend, but that's just because the backend can easily check it. If we don't promise that stricter error behavior, we don't need it.